ygot/join.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2023 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Adapted from https://cs.opensource.google/go/go/+/refs/tags/go1.20.3:src/errors/join.go
+
+// Copyright 2022 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package ygot
+
+// join returns an error that wraps the given errors.
```

Comment:
While errors.Join() is pretty specific, the name join() might not be contextualized enough in the ygot package. How about errorsJoin or joinErrors?

Reply:
Done.